### PR TITLE
feat(terraform): add new gcp postgresql checks

### LINKED ISSUE #3531: add new gcp for postgresql checks
Add checks for `database_flags`: `log_statement `, `log_hostname `, `log_min_error_statement ` and `cloudsql.enable_pgaudit` that are part of CIS 1.3. I would be happy to contribute the checks :)

### PR DESCRIPTION
**Pull request by losisin**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

Add new Google Cloud PostgreSql checks for: `log_statement `, `log_hostname `, `log_min_error_statement` and `cloudsql.enable_pgaudit`. I also takes into account default values from cloud provider. 

Fixes #3531 

### Description
These are from CIS 1.3, sections 6.2.4, 6.2.5, 6.2.7 and 6.2.9. Default values from cloud provider and from [here](https://cloud.google.com/sql/docs/postgres/flags)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by losisin** [comment]
@gruebel @JamesWoolfenden is this PR something that would be of interest?

**Comment by gruebel** [comment]
hey @losisin yeah definitely great work 🚀 I just need to find some time to go through them all 😄 next time it would be better to split the PR into 2 or 4 PRs then reviewing is also faster.

**Comment by losisin** [comment]
@gruebel next time will do!

**Review by gruebel — COMMENTED** [review]
in general changes look good. I'm not sure, how good your Python skills are, but what do you think about creating a parent class with the logic you added 4 times? then pass in the flag name and the list of valid values.

**Comment by losisin** [comment]
I agree 💯 although my python skills are very humble :). If you refer to something like [this](https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/resource/gcp/AbsGoogleComputeFirewallUnrestrictedIngress.py), I would be definitely interested. Same PR or we merge this one and then refactor all gcp postgres checks at slower pace?

**Comment by tsmithv11** [comment]
Just wanted to say thank you for including the benchmark!

File: checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlEnablePgaudit.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class GoogleCloudPostgreSqlEnablePgaudit(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure PostgreSQL database 'cloudsql.enable_pgaudit' flag is set to 'on'"
```
Comment (tsmithv11) [inline checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlEnablePgaudit.py:7]:
```suggestion
        name = "Ensure pgAudit is enabled for your GCP PostgreSQL database"
```

File: checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogHostname.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class GoogleCloudPostgreSqlLogHostname(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure PostgreSQL database 'log_hostname' flag is set to 'on'"
```
Comment (tsmithv11) [inline checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogHostname.py:7]:
```suggestion
        name = "Ensure hostnames are logged for GCP PostgreSQL databases"
```

File: checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogMinErrorStatement.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class GoogleCloudPostgreSqlLogMinErrorStatement(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure PostgreSQL database 'log_min_error_statement' flag is set to a ERROR or stricter"
```
Comment (tsmithv11) [inline checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogMinErrorStatement.py:7]:
```suggestion
        name = "Ensure the GCP PostgreSQL database log levels are set to ERROR or lower"
```

File: checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogStatement.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class GoogleCloudPostgreSqlLogStatement(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure PostgreSQL database 'log_statement' flag is set to a valid value"
```
Comment (tsmithv11) [inline checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogStatement.py:7]:
```suggestion
        name = "Ensure GCP PostgreSQL logs SQL statements"
```

**Comment by losisin** [comment]
@gruebel is the latest commit what you had in mind?

**Review by gruebel — APPROVED** [review]
great job 🚀 yeah that's what I meant 😄
you actually don't need to define the passed in parameters as a constant outside the class, directly defining them in the init method is totally enough. but no need to change

**Comment by losisin** [comment]
I really don't understand how is this possible? It's work in progress in my own branch in fork of this repository. what on earth is going on?
```
=================================== FAILURES ===================================
_______________ TestScannerRegistry.test_non_colliding_check_ids _______________
[gw1] linux -- Python 3.8.13 /home/runner/.local/share/virtualenvs/checkov-_hkiHoFg/bin/python

self = <tests.terraform.test_scanner_registry.TestScannerRegistry testMethod=test_non_colliding_check_ids>

    def test_non_colliding_check_ids(self):
        check_id_check_class_map = {}
        for (resource_type, checks) in registry.checks.items():
            for check in checks:
                check_id_check_class_map.setdefault(check.id, []).append(check)
    
        for check_id, check_classes in check_id_check_class_map.items():
>           self.assertEqual(len(set(check_classes)), 1,"collision on check_id={}".format(check_id))
E           AssertionError: 2 != 1 : collision on check_id=CKV_GCP_107

tests/terraform/test_scanner_registry.py:27: AssertionError
_________________ TestRunnerValid.test_check_ids_dont_collide __________________
[gw0] linux -- Python 3.8.13 /home/runner/.local/share/virtualenvs/checkov-_hkiHoFg/bin/python

self = <tests.terraform.runner.test_runner.TestRunnerValid testMethod=test_check_ids_dont_collide>

    def test_check_ids_dont_collide(self):
        runner = Runner()
        unique_checks = {}
        bad_checks = []
        for registry in list(runner.block_type_registries.values()):
            checks = [check for entity_type in list(registry.checks.values()) for check in entity_type]
            for check in checks:
                if check.id not in unique_checks:
                    unique_checks[check.id] = check
                elif check != unique_checks[check.id]:
                    # A single check can have multiple resource blocks it checks, which means it will show up multiple times in the registry
                    bad_checks.append(f'{check.id}: {check.name}')
                    print(f'{check.id}: {check.name}')
>       self.assertEqual(len(bad_checks), 0, f'Bad checks: {bad_checks}')
E       AssertionError: 2 != 0 : Bad checks: ['CKV_GCP_107: Cloud functions should not be public', 'CKV_GCP_107: Cloud functions should not be public']
```
Can someone help me out?

File: checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogStatement.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    'ddl',
+    'mod',
+    'all'
+]
+
+
+class GoogleCloudPostgreSqlLogStatement(AbsGooglePostgresqlDatabaseFlags):
+    def __init__(self):
+        name = "Ensure GCP PostgreSQL logs SQL statements"
+        check_id = "CKV_GCP_107"
```
Comment (tsmithv11) [inline checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogStatement.py:15]:
```suggestion
        check_id = "CKV_GCP_111"
```

CKV_GCP_107 is taken: https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/resource/gcp/CloudFunctionsShouldNotBePublic.py

Comment (losisin) [reply checkov/terraform/checks/resource/gcp/GoogleCloudPostgreSqlLogStatement.py]: oh, I see. Funny story is that I actually also work on Cloud Functions check and it started to look like episode from The Twilight Zone 😄. I've updated with master and changed check ID. Thanks @tsmithv11 !

**Review by tsmithv11 — APPROVED** [review]

**Comment by JamesWoolfenden** [comment]
Hi aleks,
Your check id has been used up by a new check that was already merged but
not yet in your branch. Merge with master and pick a new check id that
isn't in use yet and you'll be golden. This happens all the time,
especially if your pr takes a while to get merged.
James

**Pull request merged**